### ISSUE TITLE
inconsistent results when infering

### ISSUE THREAD

**Issue by melody-rain** [apache/incubator-mxnet#7398]
I trained a model with VGG16 on a dataset with accuracy of 96% on the validation set. However when I infer with the model I trained on the same validation set image by image and calculate the accuracy, I got 87%, which is quite smaller than 96%.

You know when training we make the dataset in .rec format, and for inference we read the image with cv2.imread().

What might be the reasons that I cannot get the same accuracy? All images have been resized to (224, 224) and subtracted by the mean.

FYI:
```
96%: mod.forward(batch, is_train=False), batch is from ImageRecordIter
87%: prob = mod.forward(Batch([mx.nd.array(img)])), img is from cv2.imread()
```

image preprocessing to get 96%
```
    data = mx.io.ImageRecordIter(
        data_name          = 'data',
        label_name         = 'softmax_label',
        path_imgrec        = data_val,
        label_width        = 1,
        data_shape         = data_shape,
        preprocess_threads = data_nthreads,
        batch_size         = batch_size,
        mean_r             = rgb_mean[0],
        mean_g             = rgb_mean[1],
        mean_b             = rgb_mean[2],
        )
```

image preprocessing to get 87%
```
def get_image(file_name):
    img = cv2.cvtColor(cv2.imread(file_name), cv2.COLOR_BGR2RGB)
    if img is None:
         return None

    # convert into format (batch, RGB, width, height)
    img = cv2.resize(img, (224, 224))
    img = np.swapaxes(img, 0, 2)
    img = np.swapaxes(img, 1, 2)
    img = img.astype(np.float32)
    img -= mean_img
    img = img[np.newaxis, :]
    return img

```

Thanks.

**Comment by piiswrong** [comment]
Try compare the images from imagerecorditer and get_image?

**Comment by ZhaoxiaZhang** [comment]
HI, I have met the similar problems. Have you figured this out?

**Comment by Zhaoyang-XU** [comment]
Hi, I've got the same problem for a long time.

**Comment by melody-rain** [comment]
@Zhaoyang-XU 
Hi, I suggest that you use ImageIter instead of ImageRecordIter all the time, such that you get good accuray when testing.
As for mean_X, just use imagenet‘s mean, which is (123.68,116.779,103.939). You can also calculate the mean of all the training data by yourself.

**Comment by Zhaoyang-XU** [comment]
@melody-rain Thank you, I defined my own mean. I wondered maybe it is because of the mean.

**Comment by szha** [comment]
@apache/mxnet-committers: This issue has been inactive for the past 90 days. It has no label and needs triage.

For general "how-to" questions, our [user forum](https://discuss.mxnet.io/) (and [Chinese version](https://discuss.gluon.ai/)) is a good place to get help.

**Comment by GianlucaVesentini** [comment]
I think the problem is that `ImageRecordIter` also performs data augmentation on the images (as explained [here](https://github.com/apache/incubator-mxnet/discussions/12671#discussioncomment-62256)), while if you use `cv2.imread()` you obviously have to do that on your own.

I'm facing the same issue: I trained a `mobilenetv3_small` model using a .rec file because it is clearly the preferred way, but now I'm not able to reproduce the same pre-processing operations on images before inference and this gives me completely different results.

Is there a solution to this problem?
